docs/prd/backend-provisioner.md

```
@@ -0,0 +1,1065 @@
+
+## Architecture
+
+### Backend Provisioner Registration
+
+```go
+// pkg/provisioner/backend/backend.go
+
+package backend
+
+import (
+	"github.com/cloudposse/atmos/pkg/hooks"
```

markdownlint-cli2 (0.18.1)

Hard tabs
Column: 1

(MD010, no-hard-tabs)